src/main/java/org/folio/rest/impl/OaiPmhImpl.java

```diff
@@ -53,154 +54,169 @@ public static void init(Handler<AsyncResult<Boolean>> resultHandler) {
+                                  .build();
+
+         HELPERS.get(LIST_RECORDS)
+           .handle(request, vertxContext)
+           .thenAccept(response -> {
+             if (logger.isDebugEnabled()) {
+               logger.debug("ListRecords response: " + response.getEntity());
+             }
+             asyncResultHandler.handle(succeededFuture(response));
+           })
+           .exceptionally(throwable -> {
+             asyncResultHandler.handle(succeededFuture(GetOaiRecordsResponse.respond500WithTextPlain(GENERIC_ERROR_MESSAGE)));
+             return null;
+           });
       });
   }
 
   @Override
   public void getOaiRecordsById(String id, String metadataPrefix, Map<String, String> okapiHeaders,
                                 Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    try {
-      Request request = Request.builder()
-                       .identifier(URLDecoder.decode(id, "UTF-8"))
-                       .okapiHeaders(okapiHeaders)
-                       .verb(GET_RECORD)
-                       .metadataPrefix(metadataPrefix)
-                       .identifierPrefix(buildIdentifierPrefix(okapiHeaders, vertxContext))
-                       .build();
-
-      HELPERS.get(GET_RECORD)
-        .handle(request, vertxContext)
-        .thenAccept(response -> {
-          if (logger.isDebugEnabled()) {
-            logger.debug("GetRecord response: " + response.getEntity());
-          }
-          asyncResultHandler.handle(succeededFuture(response));
-        })
-        .exceptionally(throwable -> {
+    RepositoryConfigurationHelper.getInstance().getConfiguration(okapiHeaders, vertxContext)
+      .thenAccept(v -> {
+        try {
+
+          Request request = Request.builder()
+            .identifier(URLDecoder.decode(id, "UTF-8"))
+            .okapiHeaders(okapiHeaders)
+            .verb(GET_RECORD)
+            .baseURL(getProperty(REPOSITORY_BASE_URL, vertxContext))
+            .metadataPrefix(metadataPrefix)
+            .build();
+
+          HELPERS.get(GET_RECORD)
+            .handle(request, vertxContext)
+            .thenAccept(response -> {
+              if (logger.isDebugEnabled()) {
+                logger.debug("GetRecord response: " + response.getEntity());
+              }
+              asyncResultHandler.handle(succeededFuture(response));
+            })
+            .exceptionally(throwable -> {
+              asyncResultHandler.handle(succeededFuture(GetOaiRecordsByIdResponse.respond500WithTextPlain(GENERIC_ERROR_MESSAGE)));
+              return null;
+            });
+        } catch (Exception e) {
           asyncResultHandler.handle(succeededFuture(GetOaiRecordsByIdResponse.respond500WithTextPlain(GENERIC_ERROR_MESSAGE)));
-          return null;
-        });
-    } catch (Exception e) {
-      asyncResultHandler.handle(succeededFuture(GetOaiRecordsByIdResponse.respond500WithTextPlain(GENERIC_ERROR_MESSAGE)));
-    }
+        }
```

Comment:
I think you might want an `.exceptionally(...)` here...

Reply:
added